TwoPageEBookReader/app/src/main/java/com/example/ebook_reader_sample/Book.kt

```diff
@@ -0,0 +1,202 @@
+    //TODO buildPages() supports multiple pages by cycling through pageRects instead of using only one Rect of constraints
+    fun buildPages() {
+        if (pageRects.isEmpty()) { return }
+
+        pageStrings = ArrayList()
+        var textStrings = ArrayList<String>()
+
+        var pageRectIndex = 0
+        var availableHeight = pageRects[pageRectIndex].height() - (2 * pagePadding)
+        var widthSpec = View.MeasureSpec.makeMeasureSpec(pageRects[pageRectIndex].width() - (2 * pagePadding), View.MeasureSpec.AT_MOST)
+        val heightSpec = View.MeasureSpec.makeMeasureSpec(0, View.MeasureSpec.UNSPECIFIED)
+        val paragraphStringsCopy = paragraphStrings.clone() as ArrayList<String>
+        val paragraphIt = paragraphStringsCopy.listIterator()
+
+        val textView = View.inflate(activityContext, R.layout.book_page_text_paragraph, null) as TextView
+        textView.layoutParams = LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.WRAP_CONTENT)
+        textView.textSize = fontSize.toFloat()
+        while (paragraphIt.hasNext()) {
+            val paragraphString = paragraphIt.next()
+
+            textView.text = Html.fromHtml(paragraphString)
+            textView.measure(widthSpec,heightSpec)
+            if (textView.measuredHeight > availableHeight) {
+                var splitIndex = paragraphString.lastIndexOf(". ")
+                while (splitIndex != -1) {
+                    textView.text = Html.fromHtml(paragraphString.substring(0,min(paragraphString.length,splitIndex + 2)))
+                    textView.measure(widthSpec,heightSpec)
+                    if (textView.measuredHeight < availableHeight) {
+                        textStrings.add(paragraphString.substring(0,min(paragraphString.length,splitIndex + 2)))
+                        break
+                    }
+
+                    splitIndex = paragraphString.lastIndexOf(". ", max(0, splitIndex - 1))
+                }
+
+                if (splitIndex == -1) {
+                    paragraphIt.add(paragraphString)
+                }
+                else {
+                    paragraphIt.add(paragraphString.substring(splitIndex + 2))
+                }
+                paragraphIt.previous()
+
+                pageStrings.add(textStrings)
+                textStrings = ArrayList()
+                pageRectIndex = (pageRectIndex + 1) % pageRects.size
+                widthSpec = View.MeasureSpec.makeMeasureSpec(pageRects[pageRectIndex].width() - (2 * pagePadding), View.MeasureSpec.AT_MOST)
+                availableHeight = pageRects[pageRectIndex].height() - (2 * pagePadding)
+            } else {
+                availableHeight -= textView.measuredHeight
+                textStrings.add(paragraphString)
+            }
+        }
+
+        pageStrings.add(textStrings)
+    }
+
+    private fun textStringHelper(inString: String): String {
```

Comment:
If this logic can be reused somewhere else, maybe is useful that you move its logic to a separate file so others have access.

Reply:
I'd suggest renaming the method to be clearer about _how_ it's helping

Reply:
Renamed the function. Right now it is only being used in Book.kt